panel/models/custom_html.ts

```diff
@@ -0,0 +1,155 @@
+import * as p from "@bokehjs/core/properties"
+import {ModelEvent, JSON} from "@bokehjs/core/bokeh_events"
+import {HTMLBox, HTMLBoxView} from "@bokehjs/models/layouts/html_box"
+
+import {CachedVariadicBox, set_size} from "./layout"
+
+class DOMEvent extends ModelEvent {
+  event_name: string = "dom_event"
+
+  constructor(readonly element: string, readonly event: any) {
+    super()
+  }
+
+  protected _to_json(): JSON {
+    return {model: this.origin, element: this.element, event: this.event}
+  }
+}
+
+function simplify(event: any): any {
+  const copy: any = {}
+  for (const property in event) {
+    const ptype = typeof event[property] 
+    if(ptype === "function" || ptype === 'object')
+      continue
+    copy[property] = event[property];    
+  }
+  return copy
+}
+
+function isNumeric(str: any): any {
+  if (typeof str != "string")
+    return false  
+  return !isNaN(str) && !isNaN(parseFloat(str))
```

Comment:
Better to use `Number(str)`, as it doesn't let things like `110something` through. Unfortunately it still needs special care regarding booleans, `null`, etc.

Reply:
Going to replace this with some logic that checks the DataModel property type so hopefully this goes away.